### PR TITLE
connect to OpenAI

### PR DESCRIPTION
**Pull request by Bahare09**
connect-to-OpenAI

### CODE REVIEW

**Comment by hadipournigjeh** [comment]
Well done guys

**Review by DomVinyard — APPROVED** [review]
Looks great 👍 - I will leave a couple of suggestions but happy to merge and change later

File: client/src/components/Input.js [hunk 1]
```diff
@@ -22,7 +22,9 @@ const Input = ({ onSubmit }) => {
 
 			if (response.ok) {
 				const data = await response.json();
-				onSubmit(data);
+				const correctedSentence = data.choices[0].message.content;
```
Comment (DomVinyard) [inline client/src/components/Input.js:25]: I suggest you do this: `data.choices[0].message.content` on the server side, so the `data` variable returned from the server is just the string you are interested in

File: client/src/components/Input.js [hunk 2]
```diff
@@ -22,7 +22,9 @@ const Input = ({ onSubmit }) => {
...
+				onSubmit(correctedSentence);
+				console.log(correctedSentence);
```
Comment (DomVinyard) [inline client/src/components/Input.js:27]: delete console.log

File: server/api.js [hunk 3]
```diff
@@ -9,12 +9,26 @@ router.get("/", (_, res) => {
...
+	const openai = new OpenAIApi(configuration);
 
-	// convert the input value to uppercase
-	const output = input.toUpperCase();
+	const completion = await openai.createChatCompletion({
+		model: "gpt-3.5-turbo",
+		messages: [
+			{
+				role: "user",
+				content: `correct any grammar or spelling mistakes in the sentence and offer three alternative choices : ${text}`,
```
Comment (DomVinyard) [inline server/api.js:26]: For now maybe just get it to return the best version of the sentence, something like the one we looked at on our call

```
`
You are a copywriter. Your job is to take some user text and make it better.
- Improve the grammar
- Make it more engaging 
- Make it as concise as possible without sacrificing clarity

The user typed the following:

"${text}"

Please re-write it, and make it better
`
```

Comment (DomVinyard) [reply server/api.js:26]: I think the 'alternative sentences' seems like a different feature and will require a different query?

Comment (Bahare09) [reply server/api.js:26]: The exact topic I would like to discuss tomorrow.

File: client/src/components/Input.js [hunk 4]
```diff
@@ -22,7 +22,8 @@ const Input = ({ onSubmit }) => {
 
 			if (response.ok) {
 				const data = await response.json();
-				onSubmit(data);
+				const correctedSentence = data;
+				onSubmit(correctedSentence);
```
Comment (DomVinyard) [inline client/src/components/Input.js:26]:
```suggestion
				onSubmit(data);				
```

File: server/api.js [hunk 5]
```diff
@@ -9,12 +9,37 @@ router.get("/", (_, res) => {
...
+				"${text}"
+				
+				Please re-write it, and make it better
+				`,
+			},
+		],
+	});
 
-	res.json(output);
+	res.json(completion.data.choices[0].message.content);
```
Comment (DomVinyard) [inline server/api.js:42]: what happens if there's an error?

either
- try/catch
- check beforebeforehand

```
if (!completion.data.choices.length) {
// return 404

}
```
- or both

**Review by DomVinyard — APPROVED** [review]

**Pull request merged**